### PR TITLE
MAINT: Small tidy-ups to ufunc_object.c

### PR DESCRIPTION
**Pull request by eric-wieser**
Extracted from #9022

No behavior changes, but makes the `__array_wrap__` logic a little clearer.

### CODE REVIEW

File: numpy/core/src/umath/ufunc_object.c [hunk 1]
```diff
@@ -4360,13 +4358,17 @@ ufunc_generic_call(PyUFuncObject *ufunc, PyObject *args, PyObject *kwds)
...
+            PyObject *res;
             PyObject *args_tup;
-            if (wrap == Py_None) {
-                Py_DECREF(wrap);
-                retobj[i] = (PyObject *)mps[j];
-                continue;
-            }
 
             /* Call the method with appropriate context */
             args_tup = _get_wrap_prepare_args(full_args);
```
Comment (mhvk) [inline numpy/core/src/umath/ufunc_object.c]: Not quite this PR, but while you're at it: there would seem to be no reason to call this every time in the loop; maybe preset `args_tup` to `NULL` on top, and here only set it if `NULL`? I.e.,
```
if (args_tup == NULL) {
    args_tup = _get_wrap_prepare_args(full_args);
    if (args_tup == NULL) {
        goto fail;
    }
}
```

**Review by mhvk — COMMENTED** [review]
Modulo the minor comment that is not even strictly part of this PR, this all looks OK. Definitely clearer.

File: numpy/core/src/umath/ufunc_object.c (hunk 1 above)
Comment (eric-wieser) [reply numpy/core/src/umath/ufunc_object.c]: I also considered just caching this on the `full_args` object. Something for another PR, I think.

**Comment by eric-wieser** [comment]
@mhvk, good to go? It would be nice to get this and maybe #9022 in, so that any downstream errors caused by the 1.15 `out` argument changes to `__array_wrap__` are not silenced.

File: numpy/core/src/umath/ufunc_object.c (hunk 1 above)
Comment (mhvk) [reply numpy/core/src/umath/ufunc_object.c]: OK, that's fine.

**Pull request merged**